lighthouse-core/audits/autocomplete.js

```diff
@@ -0,0 +1,106 @@
+  /** Title of a Lighthouse audit that lets the user know if there are any missing or invalid autocomplete attributes on page inputs. This descriptive title is shown to users when one or more inputs do not have autocomplete set or has an invalid autocomplete set. */
+  failureTitle: 'Input elements do not have correct attributes for autocomplete',
+  /** Description of a Lighthouse audit that lets the user know if there are any missing or invalid autocomplete attributes on page inputs. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'Autocomplete helps users submit forms quicker. To reduce user ' +
+   'effort, consider enabling autocomplete by setting the `autocomplete` ' +
+   'attribute to a valid value.' +
+  ' [Learn more](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill)',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+/** @type {string[]} */
+/** This array contains all acceptable autocomplete attributes from the WHATWG standard. More found at https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill  */
+const validAutocompleteAttributeNames = ['name', 'honorific-prefix', 'given-name',
+  'additional-name', 'family-name', 'honorific-suffix', 'nickname', 'username', 'new-password',
+  'current-password', 'one-time-code', 'organization-title', 'organization', 'street-address',
+  'address-line1', 'address-line2', 'address-line3', 'address-level4', 'address-level3',
+  'address-level2', 'address-level1', 'country', 'country-name', 'postal-code', 'cc-name',
+  'cc-given-name', 'cc-additional-name', 'cc-family-name', 'cc-number', 'cc-exp',
+  'cc-exp-month', 'cc-exp-year', 'cc-csc', 'cc-type', 'transaction-currency',
+  'transaction-amount', 'language', 'bday', 'bday-day', 'bday-month', 'bday-year',
+  'sex', 'url', 'photo', 'tel', 'tel-country-code', 'tel-national', 'tel-area-code', 'on',
+  'tel-local', 'tel-local-prefix', 'tel-local-suffix', 'tel-extension', 'email', 'impp', 'off'];
+
+class AutocompleteAudit extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'autocomplete',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['FormElements'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    const forms = artifacts.FormElements;
+    const noAutocomplete = forms.map(element => {
+      return {...element, inputs: element.inputs.filter(input => !input.autocomplete)};
+    });
+    const autocompleteInvalid = forms.map(element => {
+      return {...element, inputs: element.inputs.filter(input => {
+        for (const name of validAutocompleteAttributeNames) {
```

Comment:
This filter has no return statement and so isn't doing what you'd expect it to.

Reply:
I thought filters dont need return statements, just a boolean true or false.

when I added a return statement it didnt give me the result.

How do i properly incoperate the return statement? @connorjclark